### ISSUE TITLE
Unable to upload large video files frequently

### ISSUE THREAD

**Issue by wiseindy** [immich-app/CLI#86]
I'm using the CLI to import video files to Immich, but I've noticed that if the file > 2 GB approx, the upload fails intermittently. Upon retrying multiple times, it may or may not go through.

```
$ docker run -it --rm -v "/mediafolder:/import" ghcr.io/immich-app/immich-cli:latest upload --key ABCDE123456 --server http://192.168.1.55:2283/api --yes --recursive --delete /import/videos
```

```
Checking connectivity with Immich instance...
Server status: OK
Checking credentials...
Login status: OK
Successful authentication for user abc@example.com
Indexing local assets...
Indexing complete, found 144 local assets
Comparing local assets with those on the Immich instance...
A total of 144 assets will be uploaded to the server
Start uploading...
Upload Progress | ████████████████████████████████████████ | 100% || 144/144 || Current file [/import/videos/folder1/folder2/bigVideo.mp4]
Failed to upload 2 files  [
  {
    file: '/import/videos/folder1/folder2/bigVideo.mp4',
    reason: Error: error request aborted
        at createError (/usr/src/app/node_modules/axios/lib/core/createError.js:16:15)
        at IncomingMessage.handlerStreamAborted (/usr/src/app/node_modules/axios/lib/adapters/http.js:301:18)
        at IncomingMessage.emit (node:events:527:28)
        at IncomingMessage._destroy (node:_http_incoming:179:10)
        at _destroy (node:internal/streams/destroy:102:25)
        at IncomingMessage.destroy (node:internal/streams/destroy:64:5)
        at Socket.socketCloseListener (node:_http_client:414:11)
        at Socket.emit (node:events:539:35)
        at TCP.<anonymous> (node:net:709:12) {
      config: [Object],
      code: 'ERR_REQUEST_ABORTED',
      request: [ClientRequest],
      response: undefined,
      isAxiosError: true,
      toJSON: [Function: toJSON]
    },
    response: undefined
  },
  {
    file: "/import/videos/folder1/folder2/bigVideo2.mp4",
    reason: Error: error request aborted
        at createError (/usr/src/app/node_modules/axios/lib/core/createError.js:16:15)
        at IncomingMessage.handlerStreamAborted (/usr/src/app/node_modules/axios/lib/adapters/http.js:301:18)
        at IncomingMessage.emit (node:events:527:28)
        at IncomingMessage._destroy (node:_http_incoming:179:10)
        at _destroy (node:internal/streams/destroy:102:25)
        at IncomingMessage.destroy (node:internal/streams/destroy:64:5)
        at Socket.socketCloseListener (node:_http_client:414:11)
        at Socket.emit (node:events:539:35)
        at TCP.<anonymous> (node:net:709:12) {
      config: [Object],
      code: 'ERR_REQUEST_ABORTED',
      request: [ClientRequest],
      response: undefined,
      isAxiosError: true,
      toJSON: [Function: toJSON]
    },
    response: undefined
  }
]
```

I'm not uploading via a reverse proxy. I've directly specified the server URL above. I've also tried uploading without running the CLI in docker, i.e., I installed `immich-cli` via `npm`. I see the same error.

I've noticed that if I reboot the machine running immich and retry, 2-3 GB uploads are more likely to succeed, though anything larger still fails with the same error as above.

**Comment by wiseindy** [comment]
Nope, I just tried uploading a 5.9 GB `.mp4` video via the web UI using Firefox 113.0.2 on Windows 11.

The upload progress indicator appears and it seems to start uploading fine, but the progress bar resets after reaching 100/100 and restarts. This keeps repeating. I've left this for an hour, but the progress indicator doesn't go away and the file doesn't show up on the server. I remember seeing someone report a similar progress bar issue, but I'm not able to find it again (not sure if it was in the CLI repo or the immich repo).

This is what I see in the `immich-server` container log:

```
[Nest] 1  - 05/25/2023, 12:34:05 PM   ERROR [ExpressAdapter] Premature close
Error [ERR_STREAM_PREMATURE_CLOSE]: Premature close
    at new NodeError (node:internal/errors:399:5)
    at onclose (node:internal/streams/end-of-stream:154:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
[Nest] 1  - 05/25/2023, 12:36:15 PM   ERROR [ExpressAdapter] Premature close
Error [ERR_STREAM_PREMATURE_CLOSE]: Premature close
    at new NodeError (node:internal/errors:399:5)
    at ServerResponse.onclose (node:internal/streams/end-of-stream:154:30)
    at ServerResponse.emit (node:events:525:35)
    at emitCloseNT (node:_http_server:984:10)
    at Socket.onServerResponseClose (node:_http_server:277:5)
    at Socket.emit (node:events:525:35)
    at TCP.<anonymous> (node:net:322:12)
[Nest] 1  - 05/25/2023, 12:36:15 PM   ERROR [ExpressAdapter] Premature close
Error [ERR_STREAM_PREMATURE_CLOSE]: Premature close
    at new NodeError (node:internal/errors:399:5)
    at ServerResponse.onclose (node:internal/streams/end-of-stream:154:30)
    at ServerResponse.emit (node:events:525:35)
    at emitCloseNT (node:_http_server:984:10)
    at Socket.onServerResponseClose (node:_http_server:277:5)
    at Socket.emit (node:events:525:35)
    at TCP.<anonymous> (node:net:322:12)
```

**Comment by alextran1502** [comment]
Those errors you see are just the video stream stopping midway when you hover your mouse away from a video thumbnail, and it doesn't relate to the upload process. 

A pretty strange issue you face here, though. I try to upload an hour-long movie at 6GB, and it takes a while to encode the video, but the end still shows and is playable on the timeline 🤔. How much CPU do you allocate for the server?

**Comment by wiseindy** [comment]
I cleared all `.mp4` files in `immich/upload` and tried using the CLI again. This time the 5.9 GB file was uploaded successfully, however, I think that is likely unrelated and the issue is still there.

**Comment by wiseindy** [comment]
I'm going to try building from source and then play around with the `axios` parameters to see if I can narrow down the cause.

**Comment by wiseindy** [comment]
Update - I could be wrong, but it seems to be a server-side issue rather than client-side. I don't see anything wrong with the CLI code that does the upload.

I don't have time to build and test `immich-server` at this time, but maybe there is some sort of timeout/interruption happening server-side? I'll see if I can investigate some time later this week.

The chances of a successful upload increase dramatically after I kill all docker containers and clear the `immich/upload` dir and then start up Immich again. However, as I continue to upload many files, large files start throwing these errors more frequently.

It could be related to server resources since we're seeing this behavior. Maybe we can upload in chunks? Related discussion about using Tus: https://github.com/immich-app/immich/discussions/1674

**Comment by wiseindy** [comment]
Also, we may need better handling of failed uploads server-side, since after a bunch of failures, I see 10s or even 100s of partially uploaded files under `immich/upload` which take up storage unnecessarily and have to be manually cleaned out.

**Comment by wackyesolution** [comment]
Same issue, please fix it because for upload my 5 tb of library I had to retry more and more time.

**Comment by alextran1502** [comment]
@wackyesolution Are you uploading over local ip or dns?

**Comment by auanasgheps** [comment]
> @wackyesolution Are you uploading over local ip or dns?

Yes (we are friends)
He is using the docker container in the docs to make the upload.
Immich is running on the same server, the upload container is attached to the same docker immich network and the target is the local immich proxy.

CPU usage is about 50%-60% during upload if we disable AI jobs otherwise it's 100% lol

@wackyesolution show the full command please

**Comment by wiseindy** [comment]
This issue has been fixed for me now. (I'm using the latest version of Immich CLI and Immich server)

I think my upload errors could have been to a file system issue, though I have no way of confirming it. The hard drive I was using to store the images & videos was formatted in Windows and was connected, and mounted on Ubuntu without reformatting.

Here is what I did recently

1. Completely nuke my immich containers along with all volumes and data (I have multiple backups of my data)
2. Format my hard drive as NTFS using Ubuntu
3. Setup Immich from scratch

After doing this, I was able to upload large files (>10 GB) without a single failure or timeout. So far I have uploaded about 600 GB without any failures, so the issue was definitely with either the filesystem or my Immich setup originally.
